### ISSUE TITLE
Very slow when performing Batch Normalization Primitive on GPU.

### ISSUE THREAD

**Issue by springkim** [oneapi-src/oneDNN#696]
For the example below, comparing GPU with CPU speed shows that GPU is very slow.
https://github.com/oneapi-src/oneDNN/blob/master/examples/primitives/batch_normalization.cpp
```
std::chrono::system_clock::time_point start = std::chrono::system_clock::now();
bnorm_prim.execute(engine_stream, bnorm_args);
// Wait for the computation to finalize.
engine_stream.wait();
std::chrono::system_clock::time_point end = std::chrono::system_clock::now();
std::cout << (end - start).count() << std::endl;
```
cpu result: 9273
gpu result: 3801696

I built and used oneDNN 1.3 version (include OCL) myself.

Is it a bug or is it my fault?

## Environment
oneDNN includes hardware-specific optimizations and may behave
differently on depending on the compiler and build environment. Include
the following information to help reproduce the issue:
* CPU: i7-8700K
* OS version: Windows 10
* Compiler version: MSVC 15

**Comment by rsdubtso** [comment]
Hi @springkim. Can you please post full code of the benchmark and verbose output with DNNL_VERBOSE? Do you observe the same difference in performance when measuring more than one `execute()` call?

**Comment by springkim** [comment]
Hi @rsdubtso.
Thank you for your reply.

### Source
```cpp
#include <algorithm>
#include <cmath>
#include <iostream>
#include <string>
#include <vector>
#include<chrono>
#include "dnnl.hpp"
#include "example_utils.hpp"

using namespace dnnl;
#pragma comment(lib,"opencl.lib")

using tag = memory::format_tag;
using dt = memory::data_type;

void batch_normalization_example(dnnl::engine::kind engine_kind) {
	dnnl::engine engine(engine_kind, 0);
	dnnl::stream engine_stream(engine);
	const memory::dim N = 3, // batch size
		IC = 3, // channels
		IH = 227, // tensor height
		IW = 227; // tensor width
	memory::dims src_dims = { N, IC, IH, IW };
	memory::dims scale_shift_dims = { 2, IC };

	std::vector<float> src_data(product(src_dims));
	std::vector<float> scale_shift_data(product(scale_shift_dims));
	std::generate(src_data.begin(), src_data.end(), []() {
		static int i = 0;
		return std::cos(i++ / 10.f);
	});
	auto mid = scale_shift_data.begin() + IC;
	std::generate(scale_shift_data.begin(), mid, []() {
		static int i = 0;
		return std::sin(i++ * 2.f);
	});
	std::generate(mid, scale_shift_data.end(), []() {
		static int i = 0;
		return std::tan(i++);
	});
	auto src_md = memory::desc(src_dims, dt::f32, tag::nchw);
	auto scale_shift_md = memory::desc(scale_shift_dims, dt::f32, tag::nc);

	auto src_mem = memory(src_md, engine);
	auto scale_shift_mem = memory(scale_shift_md, engine);
	write_to_dnnl_memory(src_data.data(), src_mem);
	write_to_dnnl_memory(scale_shift_data.data(), scale_shift_mem);
	auto bnorm_d = batch_normalization_forward::desc(
		prop_kind::forward_training, src_md, 1.e-10f,
		normalization_flags::use_scale_shift
		| normalization_flags::fuse_norm_relu);
	auto bnorm_pd
		= batch_normalization_forward::primitive_desc(bnorm_d, engine);
	auto mean_mem = memory(bnorm_pd.mean_desc(), engine);
	auto variance_mem = memory(bnorm_pd.variance_desc(), engine);
	auto workspace_mem = memory(bnorm_pd.workspace_desc(), engine);
	auto bnorm_prim = batch_normalization_forward(bnorm_pd);
	std::unordered_map<int, memory> bnorm_args;
	bnorm_args.insert({ DNNL_ARG_SRC, src_mem });
	bnorm_args.insert({ DNNL_ARG_MEAN, mean_mem });
	bnorm_args.insert({ DNNL_ARG_VARIANCE, variance_mem });
	bnorm_args.insert({ DNNL_ARG_SCALE_SHIFT, scale_shift_mem });
	bnorm_args.insert({ DNNL_ARG_WORKSPACE, workspace_mem });
	bnorm_args.insert({ DNNL_ARG_DST, src_mem });


	std::chrono::system_clock::time_point start = std::chrono::system_clock::now();
	bnorm_prim.execute(engine_stream, bnorm_args);
	engine_stream.wait();
	std::chrono::system_clock::time_point end = std::chrono::system_clock::now();
	std::cout  << (end - start).count() << std::endl;

	read_from_dnnl_memory(src_data.data(), src_mem);
}

int main(int argc, char **argv) {
	batch_normalization_example(dnnl::engine::kind::gpu);
	batch_normalization_example(dnnl::engine::kind::cpu);
}
```
### Output
```
670783
5093
```

**Comment by springkim** [comment]
Using the 1.4-rc1 version has solved the above problem.
Thanks.

**Comment by vpirogov** [comment]
Thank you for letting us know.
